src/mame/machine/cxd8452aq.cpp

```diff
@@ -0,0 +1,234 @@
+// license:BSD-3-Clause
+// copyright-holders:Brice Onken
+// thanks-to:Patrick Mackinlay
+
+/*
+ * Sony CXD8452AQ WSC-SONIC3 APbus interface and Ethernet controller controller
+ *
+ * TODO:
+ *  - Error-related interrupts
+ */
+
+#include "cxd8452aq.h"
+
+#define LOG_GENERAL (1U << 0)
+#define LOG_REGISTERS (1U << 1)
+#define LOG_INTERRUPT (1U << 2)
+#define LOG_DATA (1U << 3)
+
+#define SONIC3_DEBUG (LOG_GENERAL | LOG_INTERRUPT)
+#define SONIC3_TRACE (SONIC3_DEBUG | LOG_DATA)
+
+#include "logmacro.h"
+
+DEFINE_DEVICE_TYPE(CXD8452AQ, cxd8452aq_device, "cxd8452aq", "Sony CXD8452AQ WSC-SONIC3")
+
+cxd8452aq_device::cxd8452aq_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) : device_t(mconfig, CXD8452AQ, tag, owner, clock),
+																													  device_memory_interface(mconfig, *this),
+																													  main_bus_config("main_bus", ENDIANNESS_BIG, 32, 32, 0),
+																													  sonic_config("sonic", ENDIANNESS_BIG, 32, 32, 0, address_map_constructor(FUNC(cxd8452aq_device::sonic_bus_map), this)),
+																													  m_irq_handler(*this),
+																													  m_apbus_virt_to_phys_callback(*this),
+																													  m_bus(*this, finder_base::DUMMY_TAG, -1, 64) {}
+
```

Comment:
Please just use one tab to indent initialiser lists – some of us like to have multiple windows on the screen.